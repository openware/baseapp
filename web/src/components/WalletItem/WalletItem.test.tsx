import { mount, shallow } from 'enzyme';
import { TestComponentWrapper } from 'lib/test';
import React from 'react';
import { WalletItem } from './';

const setup = (props: any) =>
    shallow(
        <TestComponentWrapper>
            <WalletItem {...props} />
        </TestComponentWrapper>
    );

describe('WalletItem #render', () => {
    it('uses styles by default', () => {
        const wrapper = setup({
            currency: 'BTC',
            name: 'Bitcoin',
            balance: '45',
            active: false,
            fixed: 8,
            iconUrl: '',
            markets: [],
            tickers: {},
            currencies: [],
        }).render();
        expect(wrapper.find('div.cr-wallet-item')).toBeDefined();
    });

    it('displays codes, balances and locked', () => {
        const wrapper = setup({
            currency: 'BTC',
            name: 'Bitcoin',
            balance: '45',
            active: false,
            fixed: 8,
            iconUrl: '',
            markets: [],
            tickers: {},
            currencies: [],
        }).render();
        expect(wrapper.find('.cr-wallet-item__description').text()).toContain('BTCBitcoin');
    });

    it('should not display locked balance if wallet does not have one', () => {
        const wrapper = setup({
            currency: 'BTC',
            name: 'Bitcoin',
            balance: '45',
            active: false,
            fixed: 8,
            iconUrl: '',
            markets: [],
            tickers: {},
            currencies: [],
        }).render();
        const balanceElement = wrapper.find('.cr-wallet-item__balance');
        expect(balanceElement.attr('class')).not.toContain('cr-wallet-item__balance-locked');
    });

    it('should render when locked = 0', () => {
        const wrapper = setup({
            currency: 'BTC',
            name: 'Bitcoin',
            balance: '45',
            active: false,
            fixed: 8,
            iconUrl: '',
            markets: [],
            tickers: {},
            currencies: [],
        }).render();
        expect(wrapper.find('.cr-wallet-item__balance-locked').text()).toEqual('');
    });

    it('should match snapshot', () => {
        const wrapper = setup({
            currency: 'BTC',
            name: 'Bitcoin',
            balance: '45',
            active: false,
            fixed: 8,
            iconUrl: '',
            markets: [],
            tickers: {},
            currencies: [],
        }).render();
        expect(wrapper).toMatchSnapshot();
    });
});
