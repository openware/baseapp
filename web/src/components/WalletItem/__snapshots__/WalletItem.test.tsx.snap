// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletItem #render should match snapshot 1`] = `
<div
  class="cr-wallet-item "
  style="display:flex;justify-content:space-between"
>
  <div
    class="cr-wallet-item__info"
  >
    <span
      class="cr-crypto-icon cr-wallet-item__icon"
    >
      <img
        alt="BTC"
      />
       
    </span>
    <div
      class="cr-wallet-item__description"
    >
      <span
        class="cr-wallet-item__description-currency"
      >
        BTC
      </span>
      <span
        class="cr-wallet-item__description-name"
      >
        Bitcoin
      </span>
    </div>
  </div>
  <span
    class="cr-wallet-item__balance"
  >
    <span>
      45
    </span>
    <span
      class="cr-decimal__opacity"
    >
      .00000000
    </span>
    <div
      class="cr-wallet-item__balance-price"
    >
      â‰ˆ $0.0000
    </div>
  </span>
</div>
`;
